deepgrp/__main__.py

```diff
@@ -307,8 +308,28 @@ def train(args: argparse.Namespace, options: dgmodel.Options):
         with open(args.parameter, "r") as file:
             parameter = dgmodel.Options.from_toml(file)
         parameter.fromdict(options.todict())
-
-        raise NotImplementedError()  # TODO:  Add training function here.
+        logdir = args.logdir
+
+        # add _LOG.xxx functions for logging
+        # include a check if preprocessed?
+        # check if logdir exists?
+        # how to load data in?
+        # specific name for trained model?
+        # load data into np.ndarray
```

Comment:
I don't think you need to check if the data is preprocessed, this can be savely assumed when loading from .npz.
Creating logdir if not exists, is a good idea, the [pathlib](https://docs.python.org/3/library/pathlib.html#pathlib.Path.mkdir) module is very nice for this.